src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,227 @@
+                            await update.TryUpdateAsync(
+                                tx,
+                                "update",
+                                updatedTime.UtcDateTime,
+                                previousTransactionDateTime
+                                );
+                            await tx.CommitAsync();
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Logger.LogError(ex, "Processing update dependency messages");
+                    }
+                }
+            }
+            else
+            {
+                Logger.LogInformation("Dependency Update Error processor is disabled because no App Configuration was available.");
+            }
+        }
+
+        public async Task<DateTimeOffset> CheckForError(DateTimeOffset previousTransactionDateTime , string whereToCreateIssue)
+        {
+            List<RepositoryBranchUpdateHistoryEntry> list = (from repo in Context.RepositoryBranchUpdateHistory
+                                                             where repo.Success == false
+                                                             where repo.Timestamp > previousTransactionDateTime.UtcDateTime
+                                                             orderby repo.Timestamp ascending
+                                                             select repo).ToList();
+            if (!list.Any())
+            {
+                return previousTransactionDateTime;
+            }
+
+            Logger.LogInformation($"Going to create the github issue");
+            try
+            {
+                await CreateGitHubIssueAsync(previousTransactionDateTime.UtcDateTime,  list[0].Repository, list[0].Branch, 
+                    list[0].Action, list[0].ErrorMessage, list[0].Method , whereToCreateIssue);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex, "Could not create a github issue for the error: " + list[0].ErrorMessage);
+            }
+            previousTransactionDateTime = list[0].Timestamp;
+            return previousTransactionDateTime;
+        }
+
+        public async Task<bool> SearchCreatedIssueAsync(DateTime transactionDateTime , string errorMessage)
+        {
+            var query = new StringBuilder();
+            HttpClient client = new HttpClient
+            {
+                BaseAddress = new Uri("https://api.github.com")
+            };
+            client.DefaultRequestHeaders.Add("User-Agent", @"Mozilla/5.0 (Windows NT 10; Win64; x64; rv:60.0) Gecko/20100101 Firefox/60.0");
+
+            query.Append($"is:issue+is:open+{"Error Message: " + errorMessage}in:title&order=desc");
+            JObject responseContent;
+            using (HttpResponseMessage response = await client.GetAsync(
```

Comment:
Why aren't we using the GitHubClient's search API? Instead of attempting to reach in anonymously to run the search? Doing it un-authenticated is going to get us a _very_ bad quota, and we should use the strongly typed search API's.

Reply:
That also means we can't see into private repositories.

Reply:
Oooh. Then the result would _also_ be strongly formatted, and we wouldn't have to be messing around with all this JSON parsing.